src/Bls12381G1KeyPair.ts

```diff
@@ -0,0 +1,506 @@
+/*
+ * Copyright 2020 - MATTR Limited
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { base64url } from "rfc4648";
+import bs58 from "bs58";
+import {
+  DEFAULT_BLS12381_G1_PUBLIC_KEY_LENGTH,
+  generateBls12381G1KeyPair,
+  blsVerify,
+  blsSign
+} from "@mattrglobal/bbs-signatures";
+import {
+  JsonWebKey,
+  KeyPairOptions,
+  KeyPairSigner,
+  KeyPairVerifier,
+  GenerateKeyPairOptions,
+  JwkKeyPairOptions,
+  BlsCurveName
+} from "./types";
+import {
+  assertBls12381G1PrivateJwk,
+  assertBls12381G1PublicJwk
+} from "./validators";
+import { convertBase64urlToBase58 } from "./utils";
+
+/**
+ * z represents the multibase encoding scheme of base58 encoding
+ * @see https://github.com/multiformats/multibase/blob/master/multibase.csv#L18
+ * @ignore
+ */
+const MULTIBASE_ENCODED_BASE58_IDENTIFIER = "z";
+
+/**
+ * 0x01 indicates the end of the leading bytes according to variable integer spec
+ * @see https://github.com/multiformats/multicodec
+ * @ignore
+ */
+const VARIABLE_INTEGER_TRAILING_BYTE = 0x01;
+
+/**
+ * 0xeb indicates a BLS 12-381 G1 public key
+ *
+ */
+const BLS12381G1_MULTICODEC_IDENTIFIER = 0xea;
+
+/**
+ * NOTE: This function isn't properly producing signatures at this time. An update
+ * needs to be made to the bbs-signatures library before this API will work properly
+ * for Bls12381G1KeyPairs.
+ *
+ * @ignore
+ * Returns an object with an async sign function for producing BBS+ signatures.
+ * The sign function is bound to the KeyPair
+ * and then returned by the KeyPair's signer method.
+ * @param key - A Bls12381G1KeyPair.
+ *
+ * @returns An object with an async function sign
+ * using the private key passed in.
+ */
+const signerFactory = (key: Bls12381G1KeyPair): KeyPairSigner => {
```

Comment:
I'd prefer if we just throw new Error("Not Implemented") for now from this function and implement it later when the underlying library actually supports it.